langchain4j-core/src/main/java/dev/langchain4j/store/embedding/EmbeddingStoreIngestor.java

```diff
@@ -122,9 +124,10 @@ private static EmbeddingModel loadEmbeddingModel() {
      * <br>
      * For the "Easy RAG", import {@code langchain4j-easy-rag} module,
      * which contains a {@code DocumentSplitterFactory} and {@code EmbeddingModelFactory} implementations.
+     * @return the token usage of the embedding process.
      */
-    public static void ingest(Document document, EmbeddingStore<TextSegment> embeddingStore) {
-        builder().embeddingStore(embeddingStore).build().ingest(document);
+    public static TokenUsage ingest(Document document, EmbeddingStore<TextSegment> embeddingStore) {
```

Comment:
I'm wondering if we shouldn't return something like an `IngestionResult` class instead, which would contain a `getTokenUsage()` method. It's an indirection, but in case we want to add more context, more details, of the ingestion, that would not be a breaking change later on as we want to add such extra info (for ex. some embedding models might truncate the text, so it could report if a segment has been truncated, etc).

Reply:
For sure, makes sense! Thanks for the review! I'll take a look on that!

Reply:
Hi @glaforge! I adjusted with an intermediary layer called `IngestResult` as you suggested. Also I included additional information included on Embeddings response that could be used :)